### ISSUE TITLE
My DW2-WiFi got added to HomeKit but they don't work as they should

### ISSUE THREAD

**Issue by DavidFG75** [bwp91/homebridge-ewelink#239]
### What issue do I have? 

When I add my two DW2-WiFi to the plug-in (as the only ones added) they work in HomeKit but intermittently (they can disappear of the Home app for more than 10 minutes), and also I can't change the room in the Home app, as well as I can't configure them as doors.
I tried to run the plug in as a separate bridge with the same instance of HomeKit. It solve the problem of the intermittence, but I still don't being able to change the configuration and location of the accessories.

Also, and that's the bigger problem for me, when I get configure the plug-in with the sensors, mi iPad stops being sync with Homebridge. It's an iPad mini running iOS 12 (it can't be updated), and the iPad is my HomeKit Hub, so without it I can't use any automatization. When I un-install the the pug-in and re-pair the Homebridge with Homekit it starts working again.

### Details of my setup:
* Using Homebridge v1.3.4 with Homebridge UI v4.41.0
* Using Homebridge as a plug-in on TrueNAS (BSD)
* NodeJS v15.14.0
* NPM v6.14.8
* I have used some versions of the plug-in, and all of them have the same problem: 6.1.2, 6.1.1, 6.0.3
* Using my  my eWeLink main account 

* Which eWeLink devices do I have that are causing issues? DW2-WiFi

**Comment by bwp91** [comment]
Hi @DavidFG75 

First of all I noticed your node version as 15.14.0, this version isn't explicitly compatible with Homebridge, I would recommend downgrading to v14.15.1.

> they can disappear of the Home app for more than 10 minutes

Do you have any logging of this? It's very unlikely that the plugin is removing devices and adding them again. And if it was it would be logged.

> and also I can't change the room in the Home app

This should be possible by long pressing on the accessory in the Home app and you can change the room in the accessory settings.

>  as well as I can't configure them as doors

It isn't possible to configure the DW2 as a door since the device itself is a sensor, not a mechanism to control a door.

> Also, and that's the bigger problem for me, when I get configure the plug-in with the sensors, mi iPad stops being sync with Homebridge. It's an iPad mini running iOS 12 (it can't be updated), and the iPad is my HomeKit Hub, so without it I can't use any automatization. When I un-install the the pug-in and re-pair the Homebridge with Homekit it starts working again.

Your logs would be helpful to see in this situation, I have a lot of users who have no problems with the DW2 device. When you mention not being in sync... do you mean just ewelink plugin or other plugins too?

**Comment by DavidFG75** [comment]
Ok, let me try downgrading to the NodeJS version that you suggest. If the problem remains, I'll send you the logs.

> This should be possible by long pressing on the accessory in the Home app and you can change the room in the accessory settings.

Yep, I know, what I mean is that even when I change the room in the accessory settings, the location didn't got saved. And actually, I have noticed that all the remaining accessories (from other plugins), stop being able to change their configuration in the Home App.

> It isn't possible to configure the DW2 as a door since the device itself is a sensor, not a mechanism to control a door.

Not as a door opener, but as a door sensor which tells you if the door is open or close. It should let me do that, actually I was able to do that when I configure the entire plugin as a separate bridge form my main "Homebridge" bridge.

> When you mention not being in sync... do you mean just ewelink plugin or other plugins too?

I mean everything on Homebridge, including other plugins, as if the connection Homekit-Homebridge was entirely broken or damaged. I had to Reset Homebridge and restore form a backup, and then re-pair to Homekit to solve it. I hope it gets solved just downgrading NodeJS, it could be causing it.

Thanks for the help, I'll let you how it outcomes. Regards.

**Comment by DavidFG75** [comment]
Update: It appear that both my iPad and the server itself, where my Homebridge is running, where making some incompatibility on the connection between HomeKit and the Homebridge.
As the Server runs Free BSD, Homebridge might not run properly. I moved everything again to a RasPi 3 and everything got solved. Also, the plugin appears to be non-compatible With iOS12, when I use another apple account (with devices running iOS 14) and I add the bridge everything works well, but if I pair the bridge with the iPad w/iOS12 the connection got lost. 
Thanks for ur help. Regards.
